api/clroni/clroni/Frame.cs

```diff
@@ -62,10 +63,17 @@ protected Frame()
         //    // Copy the memory
         //    Buffer.MemoryCopy(data.ToPointer(), frame->data, num_bytes, data_size);
         //}
+        private long mempress = 0;
+        public void addMemoryPressure(long mem)
+        {
+            mempress += mem;
+            GC.AddMemoryPressure(mem);
+        }
```

Comment:
Having this method feels hacky, but `GC.AddMemoryPressure()` needs to be called inside the class, and the Frame does not know how many bytes of unmanaged memory it has until it has been initialized by `NativeMethods.oni_read_frame()`.
Maybe we could do this with an overloaded constructor, reading an IntPtr and initializing the frame with it.

Not sure how different is to handle created frames for writing right now.